factcast-factus/src/test/java/org/factcast/factus/DefaultFactusTest.java

```diff
@@ -410,6 +448,9 @@ void withLockOnManagedProjection() {
             assertThat(locked.projection())
                     .isEqualTo(managedProjection);
 
+            // this is important; if they are not the specs for the given
+            // projection,
+            // the lock would be broken
             assertThat(locked.specs())
```

Comment:
would be easier to grasp the connection if we did 
```
            assertThat(locked.specs())
                    .isEqualTo(projector.createFactSpecs());
```

Reply:
(renamed eventapplier to projector)

Reply:
Not sure I like this, we also do
```
   when(eventApplier.createFactSpecs())
                     .thenReturn(specs);
```
and I find it awkward to use values returned by a mock for verifying...

Reply:
true, but if you look at just this line

`   assertThat(locked.specs()).isEqualTo(projector.createFactSpecs());`

it reveals the intent: the locked object must have FSpecs equal to the ones, that the projector creates.

If you just say `isEqualTo(specs) ` first of all to understand the verification, you'll first look at the definition of spec (not helpful), and than you'd need to find all the places where it is used in order to understand the semantics.

Reply:
Getting your point, but there are many reasons not to invoke methods on mocks in the test code:

- you change the number of invocations. Here, `verify(projector).createFactSpecs();` is invoked before the assertThat, but it should be legal to verify this after the assertThat but then it would crash, as it implicitly uses `times(1)`

- If for whatever reason somebody decides to reprogram the mock (e.g. `.thenReturn(specs, specs2)`) your test would break

To be honest, what we need to test here is that they are the specs for this projection, not the ones returned by the projector, and that I already mentioned in the comment above the check. If you insist, I would extend the comment to say "the ones returned by the projector", but I'm not sure if that really helps?